### PR TITLE
#159987648 Published and Favourite articles

### PR DESCRIPTION
**Pull request by iamuchejude**
#### What does this PR do?
Display favourite and published articles on user profile page.

#### Description of Task to be completed?
-- Display favourite articles.
-- Display published articles.
-- Include relevant tests.

#### How should this be manually tested?
-- Pull the latest change from this repo.
-- Run `npm run dev`.
-- Navigate to `Profile Page`.
-- Click on the published and favourite tabs.

#### What are the relevant pivotal tracker stories?
[#159987648](https://www.pivotaltracker.com/story/show/159987648)

### CODE REVIEW

File: src/reducers/index.js [hunk 1]
```diff
@@ -7,9 +7,13 @@ import category from './categories';
 import createArticle from './createArticle';
 import favorite from './favorite'
 import { oneArticleReducer, allArticleReducer, featuredReducer, recommendedReducer } from './article';
+import favorite from './favorite';
```
Comment (emmaadesile) [inline src/reducers/index.js]: Good job @iamuchejude. On line 10, you have a duplicate import of favorite reducer.

File: src/reducers/index.js [hunk 2]
```diff
@@ -7,9 +7,13 @@ import category from './categories';
...
+import { oneArticleReducer, allArticleReducer } from './article';
```
Comment (emmaadesile) [inline src/reducers/index.js]: The same thing applies here, `oneArticleReducer` and `allArticleReducer` have already been imported above

File: src/reducers/index.js (hunk 1 above)
Comment (AdmiralOnu) [inline src/reducers/index.js]: this is a duplicate of line 8

File: src/reducers/index.js (hunk 2 above)
Comment (AdmiralOnu) [inline src/reducers/index.js]: this is a duplicate of line 9

**Review by gonuoha — CHANGES_REQUESTED** [review]

**Comment by emmaadesile** [comment]
You should squash your commits. Max of 3 is good

File: src/components/UserProfileTabs/UserProfileTabs.js [hunk 3]
```diff
@@ -17,8 +18,11 @@ class UserProfileTabs extends Component {
   }
 
   componentWillMount() {
+    const { user } = this.props;
```
Comment (emmaadesile) [inline src/components/UserProfileTabs/UserProfileTabs.js:21]: This method `componentWillMount` is deprecated in react v16 and above. You should use `componentDidMount ` instead.

**Review by emmaadesile — CHANGES_REQUESTED** [review]

File: src/components/UserProfileTabs/UserProfileTabs.js (hunk 3 above)
Comment (iamuchejude) [reply src/components/UserProfileTabs/UserProfileTabs.js:21]: Okay. Thanks for the feedback
I will fix that and update the PR as soon as possible

**Review by gonuoha — APPROVED** [review]
LGTM

**Review by emmaadesile — APPROVED** [review]

File: src/reducers/favouriteArticles.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import initalState from '../store/initialState';
+import {
+  FETCH_FAVOURITE_ARTICLES_REQUEST,
+  FETCH_FAVOURITE_ARTICLES_SUCCESS,
+  FETCH_FAVOURITE_ARTICLES_FAILED,
+} from '../actionTypes/favouriteArticles';
+
+const favouriteArticles = (state = initalState.favouriteArticles, action) => {
```
Comment (johnkingzy) [inline src/reducers/favouriteArticles.js]: I think there's a typo here `initalState` instead of `initialState`

**Review by johnkingzy — CHANGES_REQUESTED** [review]
LGTM

**Review by johnkingzy — APPROVED** [review]

**Pull request merged**